packages/zoe/src/contractFacet/exit.js

```diff
@@ -0,0 +1,47 @@
+import { assert, details, q } from '@agoric/assert';
+import { E } from '@agoric/eventual-send';
+
+/**
+ * Makes the appropriate exitObj, which runs in ZCF and allows the seat's owner
+ * to request the position be exited.
+ */
+
+/** @type MakeExitObj */
+export const makeExitObj = (proposal, zoeSeatAdmin, zcfSeatAdmin) => {
+  const [exitKind] = Object.getOwnPropertyNames(proposal.exit);
+
+  /** @type {ExitObj | undefined} */
+  let exitObj;
+
+  const exitFn = () => {
+    zcfSeatAdmin.updateHasExited();
+    return E(zoeSeatAdmin).exit();
+  };
+
+  if (exitKind === 'afterDeadline') {
+    // Automatically exit the seat after deadline.
+    E(proposal.exit.afterDeadline.timer).setWakeup(
```

Comment:
The promise result is not wired in anywhere so errors are silently lost. What happens if the setWakeup doesn't work? What is the time is already past?

Reply:
That's a good point, but where would the errors be seen? For this case, there's no `tryExit()` request from the user whose result could carry the rejection.